lib/external/math.dart

```diff
@@ -40,4 +45,98 @@ ASTNode funcRand(Runtime runtime, ASTNode self, List<ASTNode> args) {
   }
 
   throw const UnexpectedTypeException('The rand method only takes [String, double, int, and bool] argument types');
+}
+
+ASTNode funcATan2(Runtime runtime, ASTNode self, List<ASTNode> args){
+  runtimeExpectArgs(args, [ASTType.AST_ANY]);
+  final y = args[0], x = args[1];
+  switch(y.type) {
+    case ASTType.AST_INT: 
+      switch(x.type) {
+        case ASTType.AST_INT: return DoubleNode()..doubleVal = atan2(y.intVal, x.intVal);
+        case ASTType.AST_DOUBLE: return DoubleNode()..doubleVal = atan2(y.intVal, x.doubleVal);
+        default:
+      } break;
+    case ASTType.AST_DOUBLE: 
+      switch(x.type) {
+        case ASTType.AST_INT: return DoubleNode()..doubleVal = atan2(y.doubleVal, x.intVal);
+        case ASTType.AST_DOUBLE: return DoubleNode()..doubleVal = atan2(y.doubleVal, x.doubleVal);
```

Comment:
Again, remember that cascade notation returns the original object.
Also, if I recall correctly, these are all redundant and can be simplified to:

```dart
final DoubleNode doubleNode = DoubleNode()..doubleVal = atan2(y.doubleVal, x.doubleVal);
```
Birb ints also save their double value in `doubleVal`